WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift

```diff
@@ -36,103 +44,87 @@ class SiteStatsDetailTableViewController: UITableViewController, StoryboardLoada
+            }
+        } else {
+            // TODO: add period receipt here
         }
-
-        tableRows.append(TableFooterRow())
-
-        return ImmuTable(sections: [
-            ImmuTableSection(
-                rows: tableRows)
-            ])
     }
 
-    func insightRow() -> ImmuTableRow? {
-
-        guard let siteStatsInsightsDelegate = siteStatsInsightsDelegate,
-            let statSection = statSection else {
-                return nil
-        }
+    func tableRowTypes() -> [ImmuTableRow.Type] {
+        return [TabbedTotalsDetailStatsRow.self,
+                TableFooterRow.self]
+    }
 
+    func storeIsFetching(statSection: StatSection) -> Bool {
         switch statSection {
+        case .insightsFollowersWordPress, .insightsFollowersEmail:
+            return insightsStore.isFetchingFollowers
         default:
-            return TopTotalsInsightStatsRow(itemSubtitle: statSection.itemSubtitle,
-                                           dataSubtitle: statSection.dataSubtitle,
-                                           dataRows: mockRows(),
-                                           siteStatsInsightsDelegate: siteStatsInsightsDelegate)
+            return false
         }
     }
 
-    func periodRow() -> ImmuTableRow? {
+    // MARK: - Table Refreshing
+
+    func refreshTableView() {
+        guard let viewModel = viewModel else {
+            return
+        }
+
+        tableHandler.viewModel = viewModel.tableViewModel()
+        refreshControl?.endRefreshing()
+    }
 
-        guard let siteStatsPeriodDelegate = siteStatsPeriodDelegate,
-        let statSection = statSection else {
-            return nil
+    @objc func refreshData() {
+        guard let statSection = statSection else {
+            return
         }
 
+        refreshControl?.beginRefreshing()
+
         switch statSection {
-        case .periodCountries:
-            return CountriesStatsRow(itemSubtitle: statSection.itemSubtitle,
-                                     dataSubtitle: statSection.dataSubtitle,
-                                     dataRows: mockRows(),
-                                     siteStatsPeriodDelegate: siteStatsPeriodDelegate)
+        case .insightsFollowersWordPress, .insightsFollowersEmail:
+            viewModel?.refreshFollowers()
         default:
-            return TopTotalsPeriodStatsRow(itemSubtitle: statSection.itemSubtitle,
-                                           dataSubtitle: statSection.dataSubtitle,
-                                           dataRows: mockRows(),
-                                           siteStatsPeriodDelegate: siteStatsPeriodDelegate)
+            refreshControl?.endRefreshing()
         }
 
-
     }
 
-    // TODO: populate table with real data.
-    // This is fake just to example the table.
-
-    func mockRows() -> [StatsTotalRowData] {
-        var dataRows = [StatsTotalRowData]()
+    func applyTableUpdates() {
+        tableView.beginUpdates()
```

Comment:
Would it be possible to use `performBatchUpdates(_:completion:)` here instead? 
The headers suggest that `beginUpdates` & `endUpdates` "will be deprecated in a future release".

Reply:
Ah, clever boy! 😄 Can do!